data-service/src/main/java/uk/gov/gchq/palisade/service/data/exception/ErrorDetails.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2019 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.palisade.service.data.exception;
+
+import uk.gov.gchq.palisade.ToStringBuilder;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+
+import static java.util.Objects.requireNonNull;
+
+public class ErrorDetails {
+
+    private Date date;
+    private String message;
+    private String details;
```

Comment:
All of these fields are null initialised by default. Please can EITHER:
1) Set them to non-null defaults.
2) Remove the no-args constructor so that fields have to be checked and initialised through the other constructor.
3) Make the no-arg constructor private.

Reply:
The no args constructor has been removed so that the fields are checked and initialised through the remaining constructor.

Reply:
This caused an issue for the tests. The no args constructor has therefore been added back in but the access has been set to private to allow tests to pass..